### PR TITLE
Update to version 21.2.5

### PR DESCRIPTION
**Pull request by matthiasdiener**
Supersedes #27 #30 #33

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is ready for review @conda-forge/mesalib

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,52 +1,105 @@
 {% set name = "mesaLib" %}
-{% set version = "18.0.0" %}
+{% set version = "21.2.5" %}
+{% set build = 0 %}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:3]: Can we remove the build number indirection?

**Comment by hmaarrfk** [comment]
This looks good.  A small nit on the recipe. But otherwise great! Did you want to use run-exports for the X11 stuff? It seems that at least libxcb has them, not sure about the others.

**Comment by matthiasdiener** [comment]
> Did you want to use run-exports for the X11 stuff? It seems that at least libxcb has them, not sure about the others.

Do you mean for pin_subpackage ? Based on https://abi-laboratory.pro/index.php?view=timeline&l=mesa , what would the right value be?

File: recipe/meta.yaml (hunk 1 above)
Comment (matthiasdiener) [reply recipe/meta.yaml]: Done in 6168c0e

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,52 +1,104 @@
...
+    - test -f $PREFIX/lib/libGLESv1_CM.1${SHLIB_EXT}  # [osx]
+    - test -f $PREFIX/lib/libGLESv2.2${SHLIB_EXT}     # [osx]
+    - test -f $PREFIX/lib/libOSMesa.8${SHLIB_EXT}     # [osx]
+    - test -f $PREFIX/lib/libGL.1${SHLIB_EXT}         # [osx]
+    - test -f $PREFIX/lib/libglapi.0${SHLIB_EXT}      # [osx]
+
+    - pkg-config --exists glesv1_cm
+    - pkg-config --exists glesv2
+    - pkg-config --exists osmesa
+    # Disable gl, egl for linux, issue with xxf86vm missing dependency, disbale OpenGL?
```
Comment (hmaarrfk) [inline recipe/meta.yaml:89]: What does this mean?

Comment (matthiasdiener) [reply recipe/meta.yaml:89]: I don't know, it is from #33 ;-)

**Comment by hmaarrfk** [comment]
More that you could remove the line `libxcb` from the `run` section since recent builds of libxcb have: https://github.com/conda-forge/libxcb-feedstock/blob/master/recipe/meta.yaml#L23

However, that is a real nit I don't care much about. I just had one question about a comment you left.

File: recipe/meta.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:89]: Should we add a new dependency? Or do yo uwant to leave this feature as one that we should include in a future build?

Comment (matthiasdiener) [reply recipe/meta.yaml:89]: I don't know. After checking again, these lines seem to come from #30. Maybe @wolfv knows more?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,52 +1,104 @@
...
+    # - pkg-config --exists gl
```
Comment (hmaarrfk) [inline recipe/meta.yaml:90]:
```suggestion
    # - pkg-config --exists gl
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,52 +1,104 @@
...
+    # - pkg-config --exists egl
```
Comment (hmaarrfk) [inline recipe/meta.yaml:91]:
```suggestion
    # - pkg-config --exists gl
    # - pkg-config --exists egl
```

File: recipe/yum_requirements.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+xorg-x11-proto-devel
+libXxf86vm-devel
+libxshmfence-devel
+mesa-libglapi
```
Comment (hmaarrfk) [inline recipe/yum_requirements.txt:4]:
```suggestion
mesa-libglapi
mesa-libGL-devel
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,52 +1,102 @@
...
-      # still too hard to get rid of python2
-      - python 2
-      - mako
-      - zlib
-      - make
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ cdt('xorg-x11-proto-devel') }}
+    - {{ cdt('libxxf86vm-devel') }}
+    - {{ cdt('libxshmfence-devel') }}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:26]:
```suggestion
    - {{ cdt('libxshmfence-devel') }}
    - {{ cdt('mesa-libgl-devel') }}  # linux
```

**Comment by hmaarrfk** [comment]
Mostly inspired by https://github.com/kovidgoyal/kitty/issues/355

**Comment by hmaarrfk** [comment]
Maybe we should just punt. This seems to be an issue for a while https://github.com/conda-forge/mesalib-feedstock/issues/16

**Comment by hmaarrfk** [comment]
Lets revert the tests for gl and refer to the issue above: https://github.com/conda-forge/mesalib-feedstock/issues/16

**Comment by hmaarrfk** [comment]
Sorry i repeated myself to avoid the use of jargon in my comment. This is a go!

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by matthiasdiener** [comment]
> Sorry i repeated myself to avoid the use of jargon in my comment. This is a go!

Great, thank you!

**Pull request merged**